packages/ui/atoms/Button/Button.twig

```diff
@@ -0,0 +1,84 @@
+{#
+/**
+ * @file
+ * Button component.
+ *
+ * @param string $label
+ * @param string $url
+ * @param array $attr
+ * @param string $icon
+ * @param string $icon_position
+ * @param 'primary'|'secondary'|'transparent' $type
+ */
+#}
+
+{# Customizable tag with a default value. #}
+{% set default_tag = url is defined or (attr is defined and attr.href is defined) ? 'a' : 'button' %}
+{% set tag = tag|default(default_tag) %}
+
+{# Default icon values #}
+{% set icon_position = icon_position|default('before') %}
+{% set icon_only = icon_only|default(false) %}
+
+{# Default attributes that could be overriden. #}
+{% set default_attributes = {
+  title: label,
+  type: tag == 'button' ? 'button' : false,
+  href: tag == 'a' and url is defined ? url : false,
+} %}
+
+{% set theme_shared = [
+  'rounded cursor-pointer transition',
+  'disabled:cursor-not-allowed',
+  {
+    'inline-block': icon is not defined,
+    'inline-flex items-center': icon is defined,
+    'px-6 py-4': not icon_only,
+    'p-4': icon_only,
+  }
+] %}
+{% set theme_primary = [
+  'text-white bg-black',
+  'hover:bg-opacity-75 disabled:bg-opacity-50'
+] %}
+
+{% set theme_secondary = [
+  'ring ring-inset ring-2 ring-black ring-opacity-25',
+  {
+    'hover:ring-opacity-100': attr is defined and attr.disabled is not defined or attr is not defined
+  }
+] %}
+
+{% set type = type|default(modifier|default('primary')) %}
+
+{# Required attributes that need to be present, i.e. classes. #}
+{% set required_attributes = {
+  class: [
+    'btn',
+    'btn--%s'|format(type),
+    attr is defined and attr.class is defined ? attr.class : [
+      theme_shared,
+      type == 'primary' ? theme_primary : '',
+      type == 'secondary' ? theme_secondary : '',
+    ]
+  ]
+} %}
+
+{# Final attributes are a merge of the defaults, the given and then the required #}
+{% set attributes = default_attributes|merge(attr|default({}))|merge(required_attributes) %}
+
+{% html_element tag with attributes %}
+  {% block content %}
+    {% if icon is defined and icon_position == 'before' %}
+      {% include '@ui/atoms/Icon/Icon.twig' with { name: icon, classes: { 'mr-3': not icon_only } } %}
+    {% endif %}
+    {% if icon_only %}
+      <span class="sr-only">{{ label }}</span>
+    {% else %}
+      {{ label }}
```

Comment:
**question**: Should label get by default an html element wrapper like `<span class="btn__label">` ?

Reply:
yeah, just having a `{% block label %}` will be enough to achieve this and keep things clean 👍